File: JavaScript/Arrays/Scaler/sum_of_all_subarrays.js
function sumOfAllSubArrays(A) {
  const n = A.length;
  let total = 0;

  for (let i = 0; i < n; i++) {
    const count = (i + 1) * (n - i);
    total += count * A[i];
  }

  return total;
}

console.log(sumOfAllSubArrays([3, 2, 4, 5]));

File: python/Arrays/Scaler/sum_of_all_subarrays.py
def sumOfAllSubArrays(A):
    n = len(A)
    total = 0

    for i in range(n):
        count = (i+1) * (n-i)
        total += (count * A[i])

    return total

# TC: O(N)
# SC: O(1)


print(sumOfAllSubArrays([3, 2, 4, 5]))
